### DISCUSSION TITLE
How to use in .net standard 2.1?

### DISCUSSION THREAD

**Discussion by Veloctor** [amerkoleci/JoltPhysicsSharp#8] [Q&A]
I need a double precision rigidbody position physics engine in Unity, this is the best choice, but Idk how to compile it in .net standard 2.1(Unity hightest support .net), there are many things not compatible with it. Is there something I missed ?

**Comment by Veloctor** [comment]
Oh I see it has been dropped. Nooooo
Is there something incompatible?

**Reply by amerkoleci** [reply]
You can try to readd it, if it works in unity I can add it back or submit PR

**Reply by Veloctor** [reply]
Hi, I have encountered some difficulties related to this, but there is no clear solution. There are also some special ideas that may limit the adaptation to within Unity (of course, precompiled branches can be used to ensure that it does not affect .NET6/7 support). I hope to exchange some infomation with you. May I contact you through Discord or other IM chats?

**Reply by amerkoleci** [reply]
I had to drop netstandard2.0 as it caused too much trouble, feel free to keep forked version.

Thanks!

**Comment by Veloctor** [comment]
> I had to drop netstandard2.0 as it caused too much trouble, feel free to keep forked version.

No problem, we just need use it in Unity, and Unity supports netstandard2.1

**Reply by amerkoleci** [reply]
Sorry, I dropped netstandard2.1 as well.

**Reply by Veloctor** [reply]
Oh... okay, I understandard. My indie aerospace simulator game need a physics engine that support real world space size simulation. That's a demo with joltphysics: [link](https://www.bilibili.com/video/BV1vV411c71H/) . Do you have any suggestion for me? I don't want to use floating origin because it need too much workaround...

**Reply by amerkoleci** [reply]
Did you had any issues with netstandard2.0 version?

**Reply by Veloctor** [reply]
Okay, it's easier than my imagine to make it suppport for Unity. [MonoPInvokeCallback(null)] is available, no need for write a lot of specific typeof(Func<param types>)s. Btw, maybe Mass/Mass in MassProperties.ScaleToMass is a bug?
